queues/c-or-cpp/circular-queue-linked-list.cpp

```diff
@@ -0,0 +1,107 @@
+// Content: Circular Queue using linked list
+// Author: Tawfik Yasser
+// Date: Mon, 8 Mar 2021
+#include <iostream>
+using namespace std;
+class Node
+{
+public:
+    int data;
+    Node *next;
+    Node() {}
+};
+class QueueLL
+{
+private:
+    Node *front;
+    Node *rear;
+    int _length;
+public:
+    QueueLL()
+    {
+        front = rear = NULL; _length = 0;
+    }
+    //Function to add new element in the queue
+    void enqueue(int value)
+    {
+        Node *newNode = new Node();
+        newNode->data = value;
+        newNode->next = NULL;
+        if(rear == NULL)
+        {
+            front = rear = newNode;
+            rear->next = front;
+        }
+        else
+        {
+            rear->next = newNode;
+            rear = newNode;
+            rear->next = front;
+        }
+      _length++;
+    }
+    //Function to remove element from the queue
+    void dequeue()
+    {
+        Node *temp = front;
+        if(front == NULL && rear == NULL){
+            cout<<"Empty Queue"<<endl;
+        }else if(front == rear){
+            front = rear = NULL;
+            delete temp;
+        }else{
+            front = temp->next;
+            rear->next = front;
+            delete temp;
+        }
+      _length--;
```

Comment:
in case the function is being called once the queue is already empty then it will display empty queue, but as the execution doesn't stop there, the value of `_length` will still be decreased to some negative value which should not happen

a simple fix is to return just after printing "Empty Queue"  from the function ( after line 48 )